src/Bitmap_cubical_complex/include/gudhi/Bitmap_cubical_complex_base.h

```diff
@@ -659,10 +659,13 @@ class Bitmap_cubical_complex_base {
   std::vector<unsigned> compute_counter_for_given_cell(std::size_t cell) const {
     std::vector<unsigned> counter;
     counter.reserve(this->sizes.size());
-    for (std::size_t dim = this->sizes.size(); dim != 0; --dim) {
-      counter.push_back(cell / this->multipliers[dim - 1]);
+    for (std::size_t dim = this->sizes.size(); dim > 1; --dim) {
+      std::size_t quot = cell / this->multipliers[dim - 1];
       cell = cell % this->multipliers[dim - 1];
+      counter.push_back(quot);
     }
+    // Split out the last iteration to save a costly division by multipliers[0]=1
```

Comment:
`multipliers[0]=1` is always true ?
Maybe we could `assert(multipliers[0] == 1);` (where it is supposed), just to be sure some future modification will not break this. If `multipliers[0]=1` is true by construction, you can forget this comment.

Reply:
As you prefer, either a `GUDHI_CHECK` everywhere you say `Split out the last iteration to save a costly division by multipliers[0]=1`, or a comment where `multipliers` is initialized to explain that the code has been optimized making the assumption that `multipliers[0]=1` when looping on multipliers values.

Reply:
I took option 2 :wink: